core/src/main/java/org/hisp/dhis/android/core/trackedentity/TrackedEntityAttributeReservedValueManager.kt

```diff
@@ -0,0 +1,406 @@
+import org.hisp.dhis.android.core.maintenance.D2ErrorComponent
+import org.hisp.dhis.android.core.organisationunit.OrganisationUnit
+import org.hisp.dhis.android.core.organisationunit.OrganisationUnitProgramLinkTableInfo
+import org.hisp.dhis.android.core.organisationunit.internal.OrganisationUnitProgramLinkStore
+import org.hisp.dhis.android.core.organisationunit.internal.OrganisationUnitStore
+import org.hisp.dhis.android.core.program.ProgramTrackedEntityAttributeTableInfo
+import org.hisp.dhis.android.core.program.internal.ProgramTrackedEntityAttributeStore
+import org.hisp.dhis.android.core.settings.GeneralSettingObjectRepository
+import org.hisp.dhis.android.core.trackedentity.internal.ReservedValueSettingStore
+import org.hisp.dhis.android.core.trackedentity.internal.TrackedEntityAttributeReservedValueQuery
+import org.hisp.dhis.android.core.trackedentity.internal.TrackedEntityAttributeReservedValueStore
+import org.hisp.dhis.android.core.trackedentity.internal.TrackedEntityAttributeStore
+import org.hisp.dhis.android.core.user.internal.UserOrganisationUnitLinkStore
+
+@SuppressWarnings("LongParameterList", "TooManyFunctions")
+@Reusable
+class TrackedEntityAttributeReservedValueManager @Inject internal constructor(
+    private val store: TrackedEntityAttributeReservedValueStore,
+    private val organisationUnitStore: OrganisationUnitStore,
+    private val trackedEntityAttributeStore: TrackedEntityAttributeStore,
+    private val programTrackedEntityAttributeStore: ProgramTrackedEntityAttributeStore,
+    private val organisationUnitProgramLinkStore: OrganisationUnitProgramLinkStore,
+    private val userOrganisationUnitLinkStore: UserOrganisationUnitLinkStore,
+    private val generalSettingObjectRepository: GeneralSettingObjectRepository,
+    private val reservedValueSettingStore: ReservedValueSettingStore,
+    private val executor: D2CallExecutor,
+    private val reservedValueQueryCallFactory: QueryCallFactory<TrackedEntityAttributeReservedValue,
+        TrackedEntityAttributeReservedValueQuery>
+) {
+    private val d2ProgressManager = D2ProgressManager(null)
+
+    /**
+     * @param attributeUid        Attribute uid
+     * @param organisationUnitUid Organisation unit uid
+     * @return Value of tracked entity attribute
+     * @see .getValue
+     */
+    suspend fun blockingGetValue(attributeUid: String, organisationUnitUid: String): String {
+        return getValue(attributeUid, organisationUnitUid)!!
+    }
+
+    /**
+     * Get a reserved value and remove it from database. If the number of available values is below a threshold
+     * (default [.FILL_UP_TO] * [.FACTOR_TO_REFILL]) it tries to download before returning a value.
+     *
+     * @param attributeUid        Attribute uid
+     * @param organisationUnitUid Organisation unit uid
+     * @return Single with value of tracked entity attribute
+     */
+
+    suspend fun getValue(attributeUid: String, organisationUnitUid: String): String? {
```

Comment:
This method should still return a `Single<String>` and remove the `suspend` modifier.

The same for other public methods, they must not change.